### PR TITLE
fix: cleanup integration folder after moving specs to e2e

### PR DESCRIPTION
**Pull request by ImCesar**
After moving the specs from `cypress/integration` to `cypress/e2e` we forgot to cleanup the old folder. This PR will now clean it up

- Closes [UNIFY-1098](https://cypress-io.atlassian.net/browse/UNIFY-1098)

### How has the user experience changed?
They don't have to manually clean up the folder or wonder if they still need it

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/data-context/src/actions/MigrationActions.ts [hunk 1]
```diff
@@ -48,6 +49,9 @@ export class MigrationActions {
     const projectRoot = this.ctx.path.join(this.ctx.currentProject)
 
     moveSpecFiles(projectRoot, specsToMove)
+    cleanUpIntegrationFolder(this.ctx.currentProject).catch((error) => {
+      throw error
```
Comment (elevatebart) [inline packages/data-context/src/actions/MigrationActions.ts:53]: Maybe push this error to the baseerror ?

**Review by elevatebart — DISMISSED** [review]
Approved with one suggestion

File: packages/data-context/src/sources/migration/utils.ts [hunk 2]
```diff
@@ -289,6 +289,25 @@ export function moveSpecFiles (projectRoot: string, specs: SpecToMove[]) {
   })
 }
 
+export async function cleanupIntegrationFolder (projectRoot: string) {
+  const integrationPath = path.join(projectRoot, 'cypress', 'integration')
+  let dirContents: string[] = []
+
+  try {
+    dirContents = await fs.readdir(integrationPath)
```
Comment (lmiller1990) [inline packages/data-context/src/sources/migration/utils.ts:297]: Should not need `try/catch`:

```ts

if (!fs.existsSync(dir)) {
  return
}

const dirContents = await fs.readdir(integrationPath)
// ... the rest ...

**Review by lmiller1990 — COMMENTED** [review]
One comment, code can be more concise - I don't think we need `try/catch` to check for existence

https://attacomsian.com/blog/nodejs-check-if-directory-exists

File: packages/data-context/src/sources/migration/utils.ts (hunk 2 above)
Comment (elevatebart) [reply packages/data-context/src/sources/migration/utils.ts:297]: @lachlan, don't do anything synchronous when dealing with files or you'll end up with race conditions and interlocks...

try catch is the recommended solution by nodejs

Comment (elevatebart) [reply packages/data-context/src/sources/migration/utils.ts:297]: https://nodejs.org/api/fs.html#fsexistspath-callback

Comment (elevatebart) [reply packages/data-context/src/sources/migration/utils.ts:297]: But you would probably want to use the version of fs that is in the ctx for consistency.

Passing it as a param to the function would do the trick.

Comment (lmiller1990) [reply packages/data-context/src/sources/migration/utils.ts:297]: Ok then, how about `if (! (await fs.exists(dir))`? Same thing.

I don't see any obvious risk of a race condition here, though, even w/ `existsSync`.

Comment (elevatebart) [reply packages/data-context/src/sources/migration/utils.ts:297]: actually according to the docs, this is the way to deal with it:

```js
  try {
    await fs.rmdir(integrationPath)
  } catch (e: any) {
    // only throw if the folder exists
    if (e.code !== 'ENOENT') {
      throw Error(`Failed to remove ${integrationPath}`)
    }
  }
```

**Review by elevatebart — DISMISSED** [review]

**Review by elevatebart — APPROVED** [review]

**Review by elevatebart — APPROVED** [review]

File: packages/data-context/src/sources/migration/utils.ts (hunk 2 above)
Comment (ImCesar) [reply packages/data-context/src/sources/migration/utils.ts]: oh awesome, this is much preferred. It also drives me nuts how much throws with fs and how it isn't really documented

**Review by ZachJW34 — APPROVED** [review]

**Pull request merged**